extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -112,7 +114,7 @@ protected Producer<String, String> setKafkaProducer()
       Properties props = new Properties();
       props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, config.getBootstrapServers());
       props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
-      props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+      props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
```

Comment:
Line no 106. String not changed to byte[]

Reply:
Compared the file with Druid 25 now. The only change is:
``` if (!eventTypes.contains(EventType.SEGMENT_METADATA) || !segmentMetadataQueue.offer(objectContainer))``` in Druid 28 vs ```if (!eventTypes.contains(EventType.SEGMENT_METADATA)) {``` in Druid 25 on line 222.
The extra part was added as part of the upstream PR: https://github.com/apache/druid/commit/4ff6026d30e4da53dc0e37bc2279d9e030773787